exp/api/v1beta1/azuremachinepool_types.go

```diff
@@ -87,6 +87,14 @@ type (
 		// SubnetName selects the Subnet where the VMSS will be placed
 		// +optional
 		SubnetName string `json:"subnetName,omitempty"`
+
+		// NetworkInterfaces to attach to the to a virtual machine.
+		// +optional
+		NetworkInterfaces []infrav1.NetworkInterface `json:"networkInterfaces,omitempty"`
+
+		// AllocatePublicIPPrefix enables the automatic creation and association of a public IP prefix for the nodes in the AzureMachinePool.
+		// +optional
+		AllocatePublicIPPrefix bool `json:"allocatePublicIPPrefix,omitempty"`
```

Comment:
We needed instance level public IPs, and I was surprised to see that AzureMachines support them, but AzureMachinePools do not. I added it in this PR because my goal with this PR was to add the network configuration flexibility we needed for our use cases. We also have a need for bring your own NICs, but that will be in a follow up PR.

Reply:
@dthorsen can we please split that feature into a separate PR? This PR is already huge and hard to review

Reply:
Agreed, doing that now